app/controllers/api/v0/events_controller.rb

```diff
@@ -16,12 +19,18 @@ def create
     render nothing: true, status: 201
   end
 
-  class KafkaData < Hash
+  class KafkaData < HashWithIndifferentAccess
+    attr_reader :schema_name
+
     def initialize(api_data:, controller:)
-      super
-      merge!(api_data)
+      super()
```

Comment:
When `KafkaData` inherited from `Hash` previously, the `super` call without parens sets the arguments to `initialize` as the default value for the hash (that which is returned for a key that does not exist).  This is what led @TomWoodward to see massive error messages dumping out the entire controller object, since `controller` is one of the arguments to `initialize`.  This behavior of `Hash` is tricky.  We changed to `HashWithIndifferentAccess` for a separate reason, a side effect of which is the `super` vs `super()` problem goes away -- this new class has a different constructor behavior, probably because `Hash`'s behavior is whacko.  Nevertheless, I am leaving in the `parens` just in case future readers don't know the different nicer behavior of `HashWithIndifferentAccess`.